mbhd-swing/src/main/java/org/multibit/hd/ui/views/wizards/send_bitcoin/SendBitcoinWizardModel.java

```diff
@@ -666,12 +669,11 @@ public void showButtonPress(HardwareWalletEvent event) {
 
             // Work out which output we're confirming (will be in same order as tx but wallet addresses will be ignored)
 
-            Optional<TransactionOutput> confirmingOutput = Optional.absent();
             do {
               // Always increment from starting position (first button request is then 0 index)
               txOutputIndex++;
-
-              if (!currentTransaction.getOutput(txOutputIndex).isMine(wallet)) {
+              log.debug("txOutputIndex:{}",txOutputIndex);
+              if (!currentTransaction.getOutput(txOutputIndex).isMine(wallet)||currentTransaction.getOutput(txOutputIndex).getAddressFromP2PKHScript(MainNetParams.get()).equals(sendRequestSummary.getDestinationAddress())||currentTransaction.getOutput(txOutputIndex).getAddressFromP2PKHScript(MainNetParams.get()).equals(sendRequestSummary.getDestinationAddress())) {
```

Comment:
Pull the new condition into a function so that is easier to understand what it is checking. Identical conditions- needs to corrected